docs/user_guides/Gufi For Coders/main.tex

```diff
@@ -0,0 +1,226 @@
+\hline
+-G \textless SQL\_aggregate\textgreater & SQL for aggregated results (no default: recommend using "SELECT * FROM entries")\\
+\hline
+-e \textless 0 or 1\textgreater & 0 for aggregate, 1 for print without aggregating (implied by -o and -O)\\
+\hline
+-m & Keep mtime and atime same on the database files \\
+\hline
+-B \textless buffer size\textgreater & size of each thread's output buffer in bytes \\
+\hline
+-w & open the database files in read-write mode instead of read only mode
+\end{tabular}
+\caption{\label{tab:widgets}Flags and Arguments}
+\end{table}
+
+\clearpage
+
+\subsection{Example Calls}
+
+gufi\_query -S "SELECT * FROM summary" ~/directory\_of\_root\_index
+\\
+gufi\_query -E "SELECT * FROM pentries" ~/directory\_of\_root\_index
+
+
+\subsection{Visualizing the Workflow}
+
+
+\begin{figure} [h]
+\centering
+\includegraphics[width=1.0\textwidth]{gufi_query.png}
+\caption{\label{fig:gufi_query}Workflow of gufi\_query}
+\end{figure}
+
+\subsection{Processdir}
+What you will quickly notice when tracing through the above diagram (Figure 3) the major functionality from this code comes from the processdir function. This function is where the -E -S and -T flags are processed. Therefore, this is where all of the SQL database accesses occur. This process is parallelized via a thread pool and is further optimized by a calculated rollupscore to ensure that we only descend into necessary sub-directories.
+
+\clearpage
+
+\subsection{querydb}
+querydb is the macro used to execute sqlite commands
+
+
+\begin{figure} [h]
+\centering
+\includegraphics[width=0.8\textwidth]{querydb.png}
+\caption{\label{fig:querydb}querydb workflow}
+\end{figure}
+
+
+\clearpage
+
+\section{gufi\_dir2index}
+
+\subsection{Outline}
+gufi\_dir2index is used in order to turn a directory into a GUFI index. This process involves creating a database template that will be copied and modified for all directories and sub-directories as gufi descends down the tree. The information of the non-directories is stored inside of the databases. Permissions are established and any resulting errors establishing them are ignored
+
+\subsection{Flags}
+
+\begin{table} [h]
+\centering
+\begin{tabular}{l|r}
+Flag & Functionality \\\hline
+-h & help manual \\
+-H & Show assigned input values \\
+-n \textless num\_threads\textgreater  & define number of threads to use \\
+-x & pull xattrs from source file-sys into GUFI \\
+-z \textless max\_level\textgreater & maximum level to go down to
+\end{tabular}
+\caption{\label{tab:widgets}Flags and Arguments}
+\end{table}
+
+\begin{figure} [h]
+\centering
+\includegraphics[width=1.0\textwidth]{gufi_dir2index.png}
+\caption{\label{fig:gufi_dir2index}gufi\_dir2index workflow}
+\end{figure}
+
+\subsection{Usage}
+gufi\_dir2index [flags] [directory to index] [location to place indexed directory]
+
+
+\section{Understanding Rollup}
+One of the predominant time savers in GUFI is the rollup feature which functions like so: \\
```

Comment:
`predominant time savers` -> `optimization`

Reply:
Made the appropriate change